libraries/rush-lib/src/logic/DependencySpecifier.ts

```diff
@@ -147,12 +152,38 @@ export class DependencySpecifier {
       if (!aliasResult.subSpec || !aliasResult.subSpec.name) {
         throw new InternalError('Unexpected result from npm-package-arg');
       }
-      this.aliasTarget = new DependencySpecifier(aliasResult.subSpec.name, aliasResult.subSpec.rawSpec);
+      this.aliasTarget = DependencySpecifier.parseWithCache(
+        aliasResult.subSpec.name,
+        aliasResult.subSpec.rawSpec
+      );
     } else {
       this.aliasTarget = undefined;
     }
   }
 
+  /**
+   * Clears the dependency specifier parse cache.
+   */
+  public static clearCache(): void {
+    dependencySpecifierParseCache.clear();
+  }
+
+  /**
+   * Parses a dependency specifier with caching.
+   * @param packageName - The name of the package the version specifier corresponds to
+   * @param versionSpecifier - The version specifier to parse
+   * @returns The parsed dependency specifier
+   */
+  public static parseWithCache(packageName: string, versionSpecifier: string): DependencySpecifier {
+    const cacheKey: string = `${packageName}\0${versionSpecifier}`;
```

Comment:
Curious: why cache the pair? Seems like the expensive part is the version specifier parse, what if the constructor cached the parse of a version specifier only? Then you can dedupe version specifiers across common versions as well. I'm also not sure if we need a new method to parse with cache in that case, which cuts down on the surface area of the change.

Reply:
Because `npmPackageArg.resolve`, which is external, takes both values. Otherwise I'd love to isolate them.

Reply:
Looks like we _could_ separate them if we pass `''` as the packageName field and then do package name validation ourselves. Will investigate as a followup.